src/main/java/core/basesyntax/lib/Injector.java

```diff
@@ -0,0 +1,31 @@
+package core.basesyntax.lib;
+
+import core.basesyntax.dao.BetDao;
+import core.basesyntax.dao.UserDao;
+import core.basesyntax.factory.BetDaoFactory;
+import core.basesyntax.factory.UserDaoFactory;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+
+public class Injector {
+
+    public static Object getInstance(Class clazz) throws IllegalAccessException,
+            InvocationTargetException, InstantiationException, NoSuchMethodException {
+        Constructor constructor = clazz.getDeclaredConstructor();
+        Object instance = constructor.newInstance();
+        Field[] fields = clazz.getDeclaredFields();
+        for (Field field : fields) {
+            if (field.getAnnotation(Inject.class) != null) {
+                field.setAccessible(true);
+                if (field.getType().equals(UserDao.class)) {
+                    field.set(instance, UserDaoFactory.getInstance());
+                }
+                if (field.getType().equals(BetDao.class)) {
+                    field.set(instance, BetDaoFactory.getInstance());
+                }
```

Comment:
refactor lines 21-26 to include `@Dao` annotation check

I would recommend to use `if-else if-else` construction and `&&` operator inside `if`

Reply:
done